src/JointTrajectoryControllerBase.cpp

```diff
@@ -578,6 +577,25 @@ void JointTrajectoryControllerBase::publishFeedback(
 
 //=============================================================================
 bool JointTrajectoryControllerBase::shouldStopExecution(std::string &message) {
```

Comment:
Changes here will get overridden by any subclass (e.g. `MoveUntilTouchTopicController`). Please move this functionality to `updateStep`.

Actually, going further, I would recommend that `shouldStopExecution` be made a `virtual` function to prevent issues like this in the future.

Reply:
Could you help me clarify this a bit? Maybe it would help to delineate what should happen in `shouldStopExecution` vs `updateStep`.

Reply:
Sorry, didn't notice that it was already virtual in the header file. You did exactly what I would have recommended: getting rid of it in the cpp file.

The issue is entirely having the check be overridden by sub-classes (like `MoveUntilTouchTopicController`), but that won't happen at this point.

If `updateStep` becomes too crowded, we can add a private function to handle things in a more modular manner.